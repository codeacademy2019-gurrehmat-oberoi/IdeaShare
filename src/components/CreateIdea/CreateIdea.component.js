import React, { Component } from 'react';
import {
  TextInput,
  Text,
  Button,
  SafeAreaView,
  KeyboardAvoidingView,
  View,
  Keyboard,
  TouchableWithoutFeedback,
} from 'react-native';
import styles from './CreateIdea.component.style';

class CreateIdea extends Component {
  state = {
    title: '',
    author: '',
    message: '',
  };

  handleTitleChange = title => this.setState({ title });

  handleAuthorChange = author => this.setState({ author });

  handleMessageChange = message => this.setState({ message });

  handleButtonPress = () => {
    this.setState({
      title: '',
      author: '',
      message: '',
    });
    this.props.navigation.navigate('Home');
  }

  render() {
    const { title, author, message } = this.state;
    return (
      <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
        <SafeAreaView style={styles.container}>
          <KeyboardAvoidingView behavior="height" keyboardVerticalOffset={100} style={styles.container}>
            <Text style={styles.title}>Create a Post!</Text>
            <TextInput placeholder="Idea Title" style={styles.textInput} onChangeText={this.handleTitleChange} value={title} />
            <TextInput placeholder="Author name (Optional)" style={styles.textInput} onChangeText={this.handleAuthorChange} value={author} />
            <TextInput placeholder="Start your idea here..." multiline style={styles.idea} onChangeText={this.handleMessageChange} value={message} />
            <View style={styles.button}>
              <Button title="Submit" onPress={this.handleButtonPress} />
            </View>
          </KeyboardAvoidingView>

        </SafeAreaView>
      </TouchableWithoutFeedback>
    );
  }
}

export default CreateIdea;
