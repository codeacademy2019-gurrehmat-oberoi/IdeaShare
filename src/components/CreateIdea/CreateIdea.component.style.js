import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'stretch',
    backgroundColor: '#F5FCFF',
    paddingLeft: 10,
    paddingRight: 10,
  },
  title: {
    fontSize: 30,
    textAlign: 'center',
    marginTop: 15,
    // top: 20,
  },
  textInput: {
    alignSelf: 'stretch',
    padding: 10,
    margin: 6,
    fontSize: 18,
  },
  idea: {
    alignSelf: 'stretch',
    // height: 400,
    borderWidth: 1,
    fontSize: 18,
    margin: 10,
    padding: 10,
  },
});

export default styles;
