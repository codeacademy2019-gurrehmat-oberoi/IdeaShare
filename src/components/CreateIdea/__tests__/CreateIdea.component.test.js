import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CreateIdea from '../CreateIdea.component';

describe('CreateIdea', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<CreateIdea />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change state title when its handler is called', () => {
    const wrapper = shallow(<CreateIdea />);
    const initialTitle = wrapper.instance().state.title;
    expect(initialTitle).toEqual('');
    wrapper.instance().handleTitleChange('test1');
    expect(wrapper.instance().state.title).toEqual('test1');
  });

  it('should change state author when its handler is called', () => {
    const wrapper = shallow(<CreateIdea />);
    const initialAuthor = wrapper.instance().state.author;
    expect(initialAuthor).toEqual('');
    wrapper.instance().handleAuthorChange('test2');
    expect(wrapper.instance().state.author).toEqual('test2');
  });

  it('should change state message when its handler is called', () => {
    const wrapper = shallow(<CreateIdea />);
    const initialMessage = wrapper.instance().state.title;
    expect(initialMessage).toEqual('');
    wrapper.instance().handleMessageChange('test3');
    expect(wrapper.instance().state.message).toEqual('test3');
  });
  it('should set state variables to empty when button is pressed', () => {
    const wrapper = shallow(<CreateIdea />);
    wrapper.instance().handleTitleChange('testTitle');
    wrapper.instance().handleAuthorChange('testAuthor');
    wrapper.instance().handleMessageChange('testMessage');
    expect(wrapper.instance().state).toEqual({
      title: 'testTitle',
      author: 'testAuthor',
      message: 'testMessage',
    });
    wrapper.instance().handleButtonPress();
    expect(wrapper.instance().state).toEqual({
      title: '',
      author: '',
      message: '',
    });
  });
});
