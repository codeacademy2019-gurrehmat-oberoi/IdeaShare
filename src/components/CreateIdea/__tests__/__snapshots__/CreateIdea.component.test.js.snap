// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIdea should render correctly 1`] = `
<RCTSafeAreaView
  accessible={true}
  emulateUnlessSupported={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingLeft": 10,
      "paddingRight": 10,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Create a Post!
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Idea Title"
      style={
        Object {
          "alignSelf": "stretch",
          "fontSize": 18,
          "margin": 6,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Author name (Optional)"
      style={
        Object {
          "alignSelf": "stretch",
          "fontSize": 18,
          "margin": 6,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Start your idea here..."
      style={
        Object {
          "alignSelf": "stretch",
          "borderWidth": 1,
          "fontSize": 18,
          "margin": 10,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
